docs/source/entities.mdx

```diff
@@ -0,0 +1,321 @@
+
+To solve this, we add a resolver to _each_ subgraph:
+
+* A resolver in Reviews to generate [representations](#entity-representations) of `Product` entities
+* A [reference resolver](#reference-resolvers) in Products to return full `Product` objects from representations
+
+### Entity representations
+
+In our example, the Reviews subgraph needs to define a resolver for its stub version of the `Product` entity. The Reviews subgraph doesn't know much about `Product`s, but fortunately, it doesn't need to. All it needs to do is return data for the fields it _does_ know about, like so:
+
+```js{3-10}:title=resolvers.js
+// Reviews subgraph
+const resolvers = {
+  Review: {
+    product(review) {
+      return {
+        __typename: "Product",
+        id: review.productID
+      };
+    }
+  },
+  // ...
+}
+```
+
+This resolver's return value is a **representation** of a `Product` entity. A representation always consists of:
+
+* A `__typename` field
+* Values for the entity's `@key` fields (`id` in this example)
+
+Because an entity can be uniquely identified by its `@key` fields, this is all the information the gateway needs to fetch additional fields for a particular `Product` object.
+
+### Reference resolvers
+
+As a reminder, here's the example query we're executing across our subgraphs:
+
+```graphql{6}
+query GetReviewsAndProducts {
+  latestReviews {
+    score
+    product {
+      id
+      price # Not defined in Reviews!
+    }
+  }
+}
+```
+
+The gateway knows it can't fetch `Product.price` from the Reviews subgraph, so _first_ it executes the following query on Reviews:
+
+```graphql{4-7}
+query {
+  latestReviews {
+    score
+    product { # List of Product representations
+      __typename
+      id
+    }
+  }
+}
+```
+
+Notice that this query _omits_ `price` but _adds_ `__typename`, even though it wasn't in the original query string! This is because the gateway knows it needs _all_ of the fields in each `Product`'s representation, including `__typename`.
+
+With these representations available, the gateway can now execute a _second_ query on the Products subgraph to fetch each product's `price`. To support this special query, the Products subgraph needs to define a **reference resolver** for the `Product` entity:
+
+```js{4-6}:title=resolvers.js
+// Products subgraph
+const resolvers = {
+  Product: {
+    __resolveReference(reference) {
+      return fetchProductByID(reference.id);
+    }
+  },
+  // ...
+}
+```
+
+> In the example above, `fetchProductByID` is a hypothetical function that fetches a `Product`'s full details from a data store based on its `id`.
+
+A reference resolver (always called `__resolveReference`) provides the gateway direct access to a particular entity's fields, without needing to use a custom query to reach that entity. To use a reference resolver, the gateway must provide a valid [entity representation](#entity-representations), which is why we created the resolver in the Reviews subgraph first!
```

Comment:
"if you use typescript and @apollo/subgraphs" :). Don't know about other subgraph server implementations, but I don't think it's part of our spec that it has to be called that way in every subgraph server. So maybe worth calling this out and saying that if you use another subgraph server, you should check their document to figure out how to declare reference resolvers.